core/src/main/java/org/parchmentmc/feather/mapping/ImmutableMappingDataContainer.java

```diff
@@ -246,18 +249,36 @@ public List<String> getJavadoc() {
             return javadoc;
         }
 
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        public ConstantData getConstant() {
+            return constant;
+        }
+
         @Override
         public boolean equals(Object o) {
             if (this == o) return true;
-            if (!(o instanceof FieldData)) return false;
-            FieldData that = (FieldData) o;
-            return getName().equals(that.getName()) && Objects.equals(getDescriptor(), that.getDescriptor())
-                    && getJavadoc().equals(that.getJavadoc());
+            if (!(o instanceof ImmutableFieldData)) return false;
+
+            ImmutableFieldData that = (ImmutableFieldData) o;
```

Comment:
This change would mean `ImmutableFieldData` would be incomparable to other `FieldData` instances (such as mutable ones). It should only check if it is an instance of `FieldData`.

Same comment on all other similar instances.